### PR TITLE
Fixed topics.regex sink validation issue for synthetic config property

### PR DESCRIPTION
**Pull request by rozza**
KAFKA-195

This is linked to #47 but I missed the usecase where users declare a `topics.regex`

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 1]
```diff
@@ -513,12 +513,11 @@ public NamespaceMapper getNamespaceMapper() {
...
         .forEach(
             (k, v) -> {
-              if (!v.errorMessages().isEmpty()) {
-                containsError.set(true);
-              }
-
               String name = topicOverrides.contains(prefix + k) ? prefix + k : k;
               if (props.containsKey(name) && !IGNORED_CONFIGS.contains(name)) {
+                if (!v.errorMessages().isEmpty()) {
+                  containsError.set(true);
```
Comment (rozza) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:519]: Only mark the error if it not being ignored.

Comment (stIncMale) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:519]: [this comment does not propose a change and does not prevent merging the PR]
The fact that we need to abuse `AtomicBoolean` as a mutable boolean holder suggests that for-each loop may be a better fit here than the `forEach` method.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java (hunk 1 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:519]: Ack

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 2]
```diff
@@ -574,11 +573,13 @@ public NamespaceMapper getNamespaceMapper() {
         .validateAll(sinkTopicOriginals)
         .forEach(
             (k, v) -> {
-              if (!v.errorMessages().isEmpty()) {
-                containsError.set(true);
+              if (!IGNORED_CONFIGS.contains(k)) {
+                if (!v.errorMessages().isEmpty()) {
+                  containsError.set(true);
```
[the remark this answers is not in the document]
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java]: Copy pasta - removed.

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 3]
```diff
@@ -525,7 +520,7 @@ public NamespaceMapper getNamespaceMapper() {
               }
             });
 
-    if (!containsError.get()) {
+    if (results.values().stream().allMatch(v -> v.errorMessages().isEmpty())) {
```
Comment (rozza) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:523]: As a for loop over entrysets is ugly and harder to read than a forEach loop - I opted to do a value check at the end.  Given validation is a startup process there is little overhead for this.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**